### PR TITLE
ENH: Give `a_min` and `a_max` default values of None in clip function.

### PR DESCRIPTION
**Pull request by Tnorm**
Further helps developers to not specify at least one of "a_min" or "a_max" arguments.

### CODE REVIEW

File: numpy/core/fromnumeric.py [hunk 1]
```diff
@@ -1950,7 +1950,7 @@ def _clip_dispatcher(a, a_min, a_max, out=None):
```
Comment (ksunden) [inline numpy/core/fromnumeric.py]: Also needs to change the `_clip_dispatcher` (line 1948) to work

**Review by Tnorm — COMMENTED** [review]
_clip_dispatcher updated

**Comment by charris** [comment]
I'm not convinced this provides much extra convenience.

**Comment by ksunden** [comment]
I will note that `ndarray.clip` provides these defaults (though the variable names themselves are different, still) So API consistency is perhaps the biggest "pro".

https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.clip.html

**Comment by Tnorm** [comment]
Would be slightly useful for one-sided clipping without facing an error. It's your choice.

**Comment by charris** [comment]
The consistency argument has convinced me.  A few more things are needed:

- A release note about the change
- An updated docstring. The arguments are now optional with a default value,
  and the release in which the change was made needs a mention. Grep for
  `.. versionchanged::` to see examples. It should be made clear that the optional part is new.

**Comment by eric-wieser** [comment]
I'm not sure consistency is applicable here if the arguments have different names. Allowing people to pass kwargs will make code less consistent due to the different spellings.

**Comment by charris** [comment]
@eric-wieser Note that the method references the function for documentation. It is unfortunate that the function has different parameter names.

**Pull request closed without merging**